cranelift/codegen/src/inst_predicates.rs

```diff
@@ -191,17 +191,18 @@ pub(crate) fn visit_block_succs<F: FnMut(Inst, Block, bool)>(
             }
 
             ir::InstructionData::BranchTable { table, .. } => {
+                let pool = &f.dfg.value_lists;
                 let table = &f.stencil.dfg.jump_tables[*table];
 
-                // The default block is reached via a direct conditional branch,
-                // so it is not part of the table. We visit the default block first
-                // explicitly, as some callers of visit_block_succs depend on that
-                // ordering.
-                visit(inst, table.default_block(), false);
-
-                for &dest in table.as_slice() {
-                    visit(inst, dest, true);
+                for dest in table.as_slice() {
+                    visit(inst, dest.block(pool), true);
```

Comment:
This change is due to the layout of `JumpTableData`, which keeps the default `BlockCall` last. The reason for mirroring the order here is so that in `machinst::lower` we can traverse the slice of `BlockCall` values directly when building up targets for branch lowering.